src/system/ZAuthModule/Controller/AccountController.php

```diff
@@ -147,55 +150,88 @@ public function lostPasswordAction(Request $request)
     }
 
     /**
-     * @Route("/lost-password/code")
+     * @Route("/lost-password/reset")
      * @Template
      * @param Request $request
      * @return array
      */
-    public function confirmationCodeAction(Request $request)
+    public function lostPasswordResetAction(Request $request)
     {
+        $redirectToRoute = 'zikulausersmodule_account_menu';
+
         if ($this->get('zikula_users_module.current_user')->isLoggedIn()) {
-            return $this->redirectToRoute('zikulausersmodule_account_menu');
+            return $this->redirectToRoute($redirectToRoute);
+        }
+
+        if (!$request->query->has('id')) {
+            $this->addFlash('error', $this->__('Your request could not be processed due to missing arguments.'));
+
+            return $this->redirectToRoute($redirectToRoute);
+        }
+
+        $lostPasswordVerificationHelper = $this->get('zikula_zauth_module.helper.lost_password_verification_helper');
+        $requestDetails = [];
+
+        try {
+            $requestDetails = $lostPasswordVerificationHelper->decodeLostPasswordId($request->query->get('id'));
+        } catch (Exception $e) {
+            $this->addFlash('error', $this->__('Your request could not be processed.') . ' ' . $e->getMessage());
+
+            return $this->redirectToRoute($redirectToRoute);
+        }
+
+        if ($requestDetails['userId'] == '' || $requestDetails['userName'] == '' || $requestDetails['emailAddress'] == '') {
+            $this->addFlash('error', $this->__('Your request could not be processed due to invalid arguments.'));
+
+            return $this->redirectToRoute($redirectToRoute);
+        }
+
+        $user = $this->get('zikula_users_module.user_repository')->findBy([
+            'uid' => $requestDetails['userId'],
+            'uname' => $requestDetails['userName'],
+            'email' => $requestDetails['emailAddress']
+        ]);
```

Comment:
I meant

> also this is from the user_repository. Anyone that has logged in since upgrade past 1.4.3 will have had their record migrated to the zauth_authentication_mapping table. So this will not work.